### PR TITLE
Feature/wopitest

### PR DESCRIPTION
**Pull request by VladMartyshevskyi**
Fixed bugs found by the wopitest.
Current status: 72 skipped/63 passed/0 failed

### CODE REVIEW

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java [hunk 1]
```diff
@@ -504,13 +513,15 @@ public String renameFile(EditorConfig config, String newTitle, String lock) thro
     }
 
     NodeImpl node = (NodeImpl) nodeByUUID(config.getFileId(), config.getWorkspace());
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Why we need NodeImpl here?

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java [hunk 2]
```diff
@@ -935,24 +952,25 @@ public String putSuggestedFile(EditorConfig config, String target, InputStream d
...
     if (config.getPermissions().contains(Permissions.USER_CAN_NOT_WRITE_RELATIVE)) {
       throw new PermissionDeniedException("User " + config.getUserId() + " cannot write relative");
     }
 
     filename = Text.escapeIllegalJcrChars(filename);
-    // Check and escape newTitle
-    if (StringUtils.isBlank(filename)) {
-      throw new IllegalFileNameException("Provided filename is illegal");
+    
+    // TODO: hadnle long filenames by trimming it. Remember about the uniqueness.
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Add more details about uniqueness: a trimmed node name should not conflict with another node name starting with the same prefix but being truncated before an unique part. Tell also that eXo JCR limit is 512 characters.

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java [hunk 3]
```diff
@@ -1064,6 +1084,12 @@ public String getMimeTypeByExtension(String extension) throws FileExtensionNotFo
         return entry.getKey();
       }
     }
+    if (extension.equals("wopitest")) {
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Not sure it's correct assumption. Let's use a default one `application/octet-stream` if it's not critical for Office services.

File: services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java [hunk 4]
```diff
@@ -62,6 +65,7 @@
 import org.exoplatform.services.log.Log;
 import org.exoplatform.services.rest.resource.ResourceContainer;
 
+// TODO: Auto-generated Javadoc
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java:68]: Cleanup it

File: services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java [hunk 5]
```diff
@@ -809,9 +819,38 @@ private Response refreshLock(EditorConfig config, String lockId) {
...
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("File not found for refresh lock", e);
+      }
+      return Response.status(Status.NOT_FOUND)
+                     .entity("{\"error\": \"" + e.getMessage() + "\"}")
+                     .type(MediaType.APPLICATION_JSON)
+                     .build();
+
+    } catch (OfficeOnlineException | RepositoryException e) {
+      LOG.error("Cannot delete file.", e);
```
Comment (pnedonosko) [inline services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java]: Here in the log message we would say more to let figure out an actual file and user, may be other details - add them (but not to the response).

**Review by pnedonosko — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java (hunk 1 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: No need now. Fixed: https://github.com/exo-addons/office-online/pull/13/commits/e93ddc3e549cd35e3bb596f7c5c6aeb0fdc73878

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java (hunk 2 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Fixed: https://github.com/exo-addons/office-online/pull/13/commits/e93ddc3e549cd35e3bb596f7c5c6aeb0fdc73878

File: services/src/main/java/org/exoplatform/officeonline/WOPIService.java (hunk 3 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/WOPIService.java]: Fixed: https://github.com/exo-addons/office-online/pull/13/commits/e93ddc3e549cd35e3bb596f7c5c6aeb0fdc73878

File: services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java (hunk 4 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java:68]: Fixed: https://github.com/exo-addons/office-online/pull/13/commits/e93ddc3e549cd35e3bb596f7c5c6aeb0fdc73878

File: services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java (hunk 5 above)
Comment (VladMartyshevskyi) [reply services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java]: Fixed: https://github.com/exo-addons/office-online/pull/13/commits/e93ddc3e549cd35e3bb596f7c5c6aeb0fdc73878

**Review by pnedonosko — APPROVED** [review]

**Pull request merged**